File: simple_automations/os_operations/javascript/batch_renamer_cli.js
// npm install argparse
const fs = require('fs');
const path = require('path');
const { ArgumentParser } = require('argparse');

const parser = new ArgumentParser({ description: 'Batch rename files in directory' }); parser.addArgument('search', { type: 'string', help: 'To be replaced text' }); parser.addArgument('replace', { type: 'string', help: 'Text to use for replacement' }); parser.addArgument('--filetype', {
  type: 'string',
  defaultValue: null,
  help: 'Only files with the given type will be renamed (e.g. .txt)',
});
parser.addArgument('--path', {
  type: 'string',
  defaultValue: '.',
  help: 'Directory path that contains the to be renamed files',
});

const args = parser.parseArgs();
console.log(args);

// rename path to dirPath because of naming conflict with the path library
const { filetype, search, replace, path: dirPath } = args;

// filter the current directory for files
const dir = fs.readdirSync(dirPath);
const files = dir.filter(f => fs.statSync(f).isFile());

console.log(`${files.length} of ${dir.length} elements are files.`);

let renamed = 0;

for (const doc of files) {
  const file = path.parse(doc);

  // skip not matching file types
  if (filetype != null && file.ext != filetype) continue;
  // skip not matching file names
  if (!file.name.includes(search)) continue;

  // rename the actual file
  fs.renameSync(doc, doc.replace(search, replace));
  renamed++;
}

console.log(`renamed ${renamed} of ${files.length} files.`);

File: simple_automations/os_operations/python/batch_renamer_cli.py
import os
import argparse

parser = argparse.ArgumentParser(description="Batch rename files in directory")

parser.add_argument("search", type=str, help="To be replaced text")
parser.add_argument("replace", type=str, help="Text to use for replacement")
parser.add_argument(
    "--filetype",
    type=str,
    default=None,
    help="Only files with the given type will be renamed (e.g. .txt)"
)
parser.add_argument(
    "--path",
    type=str,
    default=".",
    help="Directory path that contains the to be renamed files"
)

args = parser.parse_args()

# to be replaced string and file extension filter
search = args.search
replace = args.replace
type_filter = args.filetype
path = args.path

print(f"Renaming files at path {path}")

# get all files from current directory
dir_content = os.listdir(path)
path_dir_content = [os.path.join(path, doc) for doc in dir_content]
docs = [doc for doc in path_dir_content if os.path.isfile(doc)]
renamed = 0

print(f"{len(docs)} of {len(dir_content)} elements are files.")

# go through all the files and check if they match the search pattern
for doc in docs:
    # separate name from file extension
    full_doc_path, filetype = os.path.splitext(doc)
    doc_path = os.path.dirname(full_doc_path)
    doc_name = os.path.basename(full_doc_path)

    # filter for files with the right extension
    if filetype == type_filter or type_filter is None:
        # check if search text is in doc name
        if search in doc_name:
            #replace with the given text
            new_doc_name = doc_name.replace(search, replace)
            new_doc_path = os.path.join(doc_path, new_doc_name) + filetype
            os.rename(doc, new_doc_path)
            renamed += 1

            print(f"Renamed file {doc} to {new_doc_path}")

print(f"Renamed {renamed} of {len(docs)} files.")